app/src/main/java/com/nitsilchar/hp/passwordStorage/activity/LoginActivity.java

```diff
@@ -50,25 +50,24 @@ protected void onCreate(Bundle savedInstanceState) {
         str_getEmail = SplashActivity.sh.getString("email", null);
         str_getPass = SplashActivity.sh.getString("password", null);
         login = (Button) findViewById(R.id.btn_login);
-        register=(Button)findViewById(R.id.btn_reg);
+        toreg = (Button) findViewById(R.id.toreg);
         edt_Email = (EditText) findViewById(R.id.edt_email);
         edt_Password = (EditText) findViewById(R.id.edt_password);
         cardView1=(CardView)findViewById(R.id.card1);
         cardView2=(CardView)findViewById(R.id.card2);
         progressBar=(ProgressBar)findViewById(R.id.progressBar);
         b=splashActivity.containsPass("password");
         if(b==true){
-            register.setVisibility(View.INVISIBLE);
-            cardView2.setVisibility(View.INVISIBLE);
         }
         login.setOnClickListener(this);
-
-        register.setOnClickListener(new View.OnClickListener() {
+        toreg.setOnClickListener(this);

```

Comment:
Why do we have two `onClickListener`s set for `toreg`? This causes unpredicatable behavior, please remove the reference to `this` and go with the anonymous inner class route.

Reply:
Removed the extra onClickListener